src/net_processing.cpp

```diff
@@ -724,20 +712,17 @@ bool ProcessMessage(CNode *pfrom, std::string strCommand, CDataStream &vRecv, in
 
         pfrom->ReadConfigFromXVersion();
 
-        pfrom->PushMessage(NetMsgType::XVERACK);
-        pfrom->state_incoming = ConnectionStateIncoming::READY;
-        handleAddressAfterInit(pfrom);
-        enableSendHeaders(pfrom);
-        enableCompactBlocks(pfrom);
+        pfrom->PushMessage(NetMsgType::XVERACK_OLD);
+        // handleAddressAfterInit(pfrom);
+        // enableSendHeaders(pfrom);
+        // enableCompactBlocks(pfrom);
     }
-    else if (strCommand == NetMsgType::XVERACK)
+    else if (strCommand == NetMsgType::XVERACK_OLD)
     {
-        if (!ensureConnectionState(strCommand, ConnectionStateIncoming::ANY, ConnectionStateOutgoing::READY, pfrom))
-            return false;
-
         // This step done after final handshake
-        CheckAndRequestExpeditedBlocks(pfrom);
+        // CheckAndRequestExpeditedBlocks(pfrom);
```

Comment:
checkandrequest is commented out? don't we still need it?

Reply:
it is still called under the verack message.